src/ansys/dpf/post/index.py

```diff
@@ -165,9 +165,24 @@ class ResultsIndex(Index):
     def __init__(
         self,
         values: List[str],
+        units: Union[List[Union[str, None]], None] = None,
     ):
         """Initiate this class."""
-        super().__init__(name=ref_labels.results, values=values, scoping=None)
+        result_values = values
+        if units is not None:
+            for i, unit in enumerate(units):
+                result_values[i] += f" ({unit})" if unit is not None else ""
+            if len(units) < len(values):
+                units.extend([None] * (len(values) - len(units)))
+        else:
+            units = [None] * len(values)
```

Comment:
> Probably we want to avoid writing `"()"` if `unit` is `None`. :)
> 
> And personally, I read better the `if` before the `for`. But this is a nit pick :).

Hi @germa89! Thx for the review! Just to nit pick too, the original line 174 does actually write `""` in `unit` is `None` ;)

Also, the `if` after the `for` is because the operation of _adding the unit to the string rep_ if not needed for `None` values of units, which is all we are adding in the `if`. 
Thus having the `if` before the `for` might be a bit more readable, but is adds unnecessary operations. What I will do is add comments so it is more readable because you are right, it is not really readable.